sources/zarr/large_image_source_zarr/__init__.py

```diff
@@ -653,6 +648,111 @@ def addTile(self, tile, x=0, y=0, mask=None, axes=None, **kwargs):
                 self.levels = int(max(1, math.ceil(math.log(max(
                     self.sizeX / self.tileWidth, self.sizeY / self.tileHeight)) / math.log(2)) + 1))
 
+    def addAssociatedImage(self, image):
```

Comment:
We should change this method signature to `def addAssociatedImage(self, image, imageKey)`, where imageKey is a string that we store and can use to identify what this associated image is used for.

Reply:
Note that you couldn't set the imageKey in the vips sink (we could go back and add it eventually).

Reply:
We don't currently read in the imageKey when finding associated images in zarr files.  If we used the key as the zarr group OR used associated_<num>, then we could use the group name on read (this means changing https://github.com/girder/large_image/blob/master/sources/zarr/large_image_source_zarr/__init__.py#L323-L324 and associated areas).

Reply:
Ok, I made this change in [8976f3d](https://github.com/girder/large_image/pull/1508/commits/8976f3df1790217e1b45ca6466907488ff7c3cc0). We now store `self._associatedImages` as a dict.